File: app/src/main/java/dev/olog/basil/presentation/main/RecipeTitleAdapter.java
package dev.olog.basil.presentation.main;

import androidx.databinding.ViewDataBinding;
import dev.olog.basil.BR;
import dev.olog.basil.R;
import dev.olog.basil.presentation.base.BaseAdapter;

public class RecipeTitleAdapter extends BaseAdapter<String> {

    @Override
    protected void initViewHolderListeners(ViewDataBinding binding) {

    }

    @Override
    protected void bind(ViewDataBinding binding, String item, int position) {
        binding.setVariable(BR.item, item);
    }

    @Override
    protected int provideLayoutId() {
        return R.layout.item_recipe_title;
    }
}

File: presentation/src/main/java/dev/olog/basil/presentation/main/RecipeTitleAdapter.kt
package dev.olog.basil.presentation.main

import dev.olog.basil.presentation.base.BaseAdapter
import dev.olog.basil.presentation.base.DataBoundViewHolder
import kotlinx.android.synthetic.main.item_recipe_title.view.*

class RecipeTitleAdapter : BaseAdapter<TitleModel>() {

    override fun initViewHolderListeners(viewHolder: DataBoundViewHolder, viewType: Int) {

    }

    override fun bind(holder: DataBoundViewHolder, item: TitleModel, position: Int) {
        holder.view.apply {
            title.text = item.title
        }
    }
}
